docs/src/architecture/08_concepts/cardano_chain_follower/follower_instances.md

```diff
@@ -0,0 +1,122 @@
+
+1. App 1 subscribes to genesis.
+2. Connection to Node occurs, and block 1 is received, and sent to App 1.
+3. App 2 subscribes to genesis.
+4. The first connection can not be used because we are already past genesis on that stream, so we need to create a second stream.
+
+The problem here is we have two streams, but they are only 1 block apart.  
+Further, based on timing it may be that either exceeds the other depending on how fast their individual connection to the relay is.
+This is not ideal.
+
+There are a number of ways this can be solved.
+
+1. Use a block cache for recent blocks, so that subscriptions that are close in time can use the same stream by
+   re-using blocks from the cache.
+2. Delay when we connect a stream to allow for the maximum number of subscribers.
+3. Optimizing the streams on the fly.
+    * For example if 2 streams are at different points, but they could catch up and become equal.
+    the moment they become equal one of the streams is closed and all block data is now taken from a single stream.
+    * This is likely to occur, for example a follower from Tip and Genesis.
+    Once the Genesis follower has reached tip we now have two followers from Tip when only 1 is required.
+4. An ideal solution may need multiple techniques.
+
+*Note: There may be other techniques and these are just some obvious solutions to the issue.*
+
+For the initial implementation of the Follower, we will employ technique 2 and a limited version of Technique 3 ONLY.
+
+#### Delaying starting of the streams
+
+When a stream is subscribed, and it results in a new stream, then the connection to that relay node will not happen for 5 seconds.
+Any further subscriptions to the same stream can then be added to the pending connection.
+Once the connection is made, then further subscriptions will only be added to it if they are valid for that connection.
+Otherwise a second connection is established, again with its own 5 second start delay.
+
+This delay occurs regardless of where the stream is to start from Tip, Genesis or a particular block.
+
+#### Optimizing the streams
+
+If we have two followers for a particular network, and they are BOTH on TIP,
+then one is stopped and all subscribers for it are moved to the other.
+
+This needs to check that not only are they both reported to be on TIP, but that the TIPs are identical.
+
+We do not need to do any other stream optimization at this time.
+
+The purpose of this is to ensure that once followers have fully synced we are not running excessive followers
+that are doing the same work and reporting the same events.
+
+### Handling "Continue" connections
+
+There is an option to listen to a stream from "Continue" but what does "Continue" actually mean?
+If there are multiple streams it could match on, which one does it match?
+
+We will use the rule that if "Continue" is specified, it means "Follow the blockchain with the earliest blocks".
+
```

Comment:
Storing the point each follower is at would be used at this point as well as my comment above. I guess it would also allow the runtime extension to find a follower for a module on a "continue" request.

We could also make it reuse the logic I mentioned above for not only finding the follower with the earliest blocks but to wait and, possibly, spawn a follower for conitnuing at the point it was stopped. But this is just a thought.

Reply:
Yes each follower will need to know where its at for "continue" to work in its simplest case.
See above for not needing to do these optimizations yet, except for probably tracking where a follower is at so we can properly implement "continue" in a simple way initially.